lld/test/COFF/dependentflags.test

```diff
@@ -29,3 +30,4 @@ FAIL: lld-link: error: /dependentloadflag: invalid argument: zz
 FAIL-RANGE: lld-link: error: /dependentloadflag: invalid argument: 0xf0000
 FAIL-NOARG: lld-link: error: /dependentloadflag: no argument specified
 
+WARN-NOBASE: lld-link: warning: _load_config_used not found, /delayloadflag will have no effect
```

Comment:
Please add a test to check that no warnings are emitted if `_load_config_used` exists.

Reply:
And to be really clear, ideally that test should have the `_load_config_used` symbol in a static library, so it’s not explicitly referenced or pulled in, other than lld doing it implicitly at the end.

Reply:
> Please add a test to check that no warnings are emitted if `_load_config_used` exists.

Good point, added.

> And to be really clear, ideally that test should have the `_load_config_used` symbol in a static library, so it’s not explicitly referenced or pulled in, other than lld doing it implicitly at the end.

Just to confirm. It'd be enough to turn `ldcfg.obj` into a static library and directly pass that library to LLD?

Reply:
FWIW, I also felt a little skeptic that this would be enough, as we're only scanning the files that are mentioned directly on the command line:

https://github.com/llvm/llvm-project/blob/llvmorg-19.1.4/lld/COFF/Driver.cpp#L2134-L2169

As technically, the `_load_config_used` struct can be brought in at a later stage as well, here: https://github.com/llvm/llvm-project/blob/llvmorg-19.1.4/lld/COFF/Driver.cpp#L2397-L2448

I tried to adjust the testcase even further, so that we have two files on the command line - an entry point object, and a library. The entry point object references the library, and the object in the library has got an embedded `/defaultlib:` directive, pointing to another library, which has got the `_load_config_used` struct. But even then, the warning works as expected - not warning for this.

I also tested with real MSVC libraries (where there's usually a somewhat deep nested structure of `/defaultlib:` directives between msvcrt/ucrt/vcruntime etc), but there, the `_load_config_used` is also in a toplevel library, and linking a regular hello world, with `/dependentloadflag:0`, doesn't produce any warning here.

So with that, I think this may cover enough of the practical cases and keeps the code simple. Or what do you think @alvinhochun?

Reply:
Thanks for testing this!
Should I cherry-pick 1a5c668 to this PR?

Reply:
Not sure - what does @alvinhochun think? It makes the testcase more complicated, but also closer to the real world scenario.

Reply:
The test in that commit `RUN: lld-link %t.entry.obj /libpath:%t-libdir %t.ldcfg-defaultlib.lib /out:%t.exe /entry:entry /subsystem:console /dependentloadflag:0x801 2>&1 | FileCheck %s --allow-empty --check-prefix NO-WARN` does not have `/nodefaultlib`, would that not pull in the default CRT libs and invalidate the test?

Reply:
I needed to remove the `/nodefaultlib` to make it actually do the thing I wanted to test, having `%t.ldcfg-defaultlib.lib` contain a `/defaultlib:` directive, that pulls in another library - with `/nodefaultlib` it would ignore those.

Linking without `/nodefaultlib` doesn't pull in the default CRT libs automatically, only if the object files tells it to. In this case, `%S/Inputs/precomp-a.obj` contains such directives, but I switched that to a different object file for the entry, which doesn't reference `LIBCMT.LIB` and such.